internal/entitlement/metered/entitlement.go

```diff
@@ -23,6 +24,12 @@ type Entitlement struct {
 
 	// UsagePeriod defines the recurring period for usage calculations.
 	UsagePeriod entitlement.UsagePeriod `json:"usagePeriod,omitempty"`
+
+	// CurrentPeriod defines the current period for usage calculations.
+	CurrentUsagePeriod api.Period `json:"currentUsagePeriod,omitempty"`
```

Comment:
- lets not reference api in the core implementations
- shouldn't this be optional?